src/memory_manager.rs

```diff
@@ -169,6 +169,14 @@ pub fn start_worker<VM: VMBinding>(
 /// * `tls`: The thread that wants to enable the collection. This value will be passed back to the VM in
 ///   Collection::spawn_worker_thread() so that the VM knows the context.
 pub fn enable_collection<VM: VMBinding>(mmtk: &'static MMTK<VM>, tls: VMThread) {
+    // initialize a list of perf events for work thread initialization
+    #[cfg(feature = "perf")]
+    {
+        use crate::scheduler::stat::PERF_EVENTS;
+        use crate::scheduler::work_counter::parse_perf_events;
+        let mut w = PERF_EVENTS.write().unwrap();
+        *w = parse_perf_events(&mmtk.options.perf_events)
+    }
     mmtk.scheduler.initialize(mmtk.options.threads, mmtk, tls);
```

Comment:
You can change the first parameter in `initialize()` to be `Options`, and do the initialization in `scheduler.initialize()`.